torch/csrc/jit/script/compiler.cpp

```diff
@@ -2772,32 +2772,46 @@ struct to_ir {
   }
 };
 
+struct MethodResolver : public Resolver {
+  explicit MethodResolver(
+      const Resolver* otherResolver,
+      const std::unordered_map<std::string, Function*>& functionTable)
+      : otherResolver_(otherResolver), functionTable_(functionTable) {}
+
+  std::shared_ptr<SugaredValue> resolveValue(
+      const std::string& name,
+      Function& m,
+      const SourceRange& loc) const override {
+    auto it = functionTable_.find(name);
+    if (it != functionTable_.end()) {
+      return std::make_shared<MethodValue>(c10::nullopt, *it->second);
+    }
+    return otherResolver_->resolveValue(name, m, loc);
+  }
+
+ private:
+  const Resolver* otherResolver_;
```

Comment:
This fails to forward type resolution calls to otherResolver_, which I think is wrong. I also wonder if this is a good pattern. What happens if we add another method to Resolver? How would someone know to update this proxy.

Reply:
Maybe it is better to just move the function_table_ logic into to_ir rather than try to push it into the resolver.

Reply:
Now I see that type resolution does not use this yet, so I assume it is not a bug yet, but it seems like a likely place where a bug could happen.

Reply:
Yeah, wrapping the resolver will probably be fragile. I'll change

Reply:
Actually, I changed my mind. Since `Resolver` is an abstract virtual, adding new methods to it will require implementations by inheritors, so it'll be hard to actually introduce a bug. Passing a function_table around feels unnecessary when we already have a way to do "look up this name in the outer environment".